tests/test_text.py

```diff
@@ -0,0 +1,464 @@
+# -*- coding: utf-8 -*-
+# Copyright FMR LLC <opensource@fidelity.com>
+# SPDX-License-Identifier: GNU GPLv3
+import random
+import selectors
+import pandas as pd
+from feature.selector import Selective, SelectionMethod
+from test_base import BaseTest
+from textwiser import TextWiser, Embedding, Transformation
+
+
+class TestText(BaseTest):
+
+    """
+        For details about test parameters see TextBased in Selector.py
+    """
+    ################################################
+    ########## Tests for random selection ##########
+    ################################################
+    def test_text_based_random_num_feature(self):
+        data = pd.DataFrame(
+            {"item1": ["this is a sentences with more common words and more words to increase frequency"],
+             "item2": ["second one in list of more frequent sentences with some repeated words"],
+             "item3": ["a word for more complexity and longer length"],
+             "item4": ["sentence with a lot of repeated common words and more words to increase frequency"],
+             "item5": ["more frequent words with a valid vector and more words to increase frequency"]})
+        labels = pd.DataFrame({"item1": [0, 1, 0, 0, 0], "item2": [1, 0, 0, 1, 0], "item3": [0, 0, 1, 0, 1],
+                               "item4": [0, 1, 1, 0, 1], "item5": [0, 1, 0, 0, 1]})
+
+        method_unicost = SelectionMethod.TextBased(num_features=3,
+                                                   optimization_method="random",
+                                                   cost_metric="unicost",
+                                                   trials=1)
+
+        method_diverse = SelectionMethod.TextBased(num_features=3,
+                                                   optimization_method="random",
+                                                   cost_metric="diverse",
+                                                   trials=1)
+
+        selector_unicost = Selective(method_unicost)
+        selector_unicost.fit(data, labels)
+        selected_features_unicost = set(selector_unicost.transform(data).columns)
+
+        selector_diverse = Selective(method_diverse)
+        selector_diverse.fit(data, labels)
+        selected_features_diverse = set(selector_diverse.transform(data).columns)
+
```

Comment:
If you could add a quick comment above the assertion about the exception, it makes it a lot easier to understand what the test is doing. For example, here it would be sth like "random should result the same selection with uni/diverse cost"

Reply:
Added.

Reply:
A comment added for each test